alacritty_terminal/src/term/mod.rs

```diff
@@ -2129,8 +2122,10 @@ impl<T: EventListener> Handler for Term<T> {
         trace!("Attempting to pop title from stack...");
 
         if let Some(popped) = self.title_stack.pop() {
-            trace!("Title '{}' popped from stack", popped);
-            self.set_title(&popped);
+            trace!("Title '{:?}' popped from stack", popped);
+
+            self.set_title(popped.as_ref().unwrap_or(&self.default_title).clone());
+            self.title = popped;
```

Comment:
I don't really like that we change to `Default title` and then back to `None`, maybe we should use `Option<String>` in `set_title`.

Reply:
Yeah, I've had that temporarily. It's probably a good idea. Though it's unfortunate since this is the only place where we need it and it would make things easier not to have it everywhere elese.